hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/MessagesPublishingStartupValidationHook.java

```diff
@@ -0,0 +1,121 @@
+package pl.allegro.tech.hermes.frontend.server;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.Fallback;
+import net.jodah.failsafe.RetryPolicy;
+import net.jodah.failsafe.event.ExecutionAttemptedEvent;
+import net.jodah.failsafe.function.CheckedConsumer;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.common.config.ConfigFactory;
+import pl.allegro.tech.hermes.common.config.Configs;
+import pl.allegro.tech.hermes.common.hook.Hook;
+import pl.allegro.tech.hermes.common.hook.ServiceAwareHook;
+import pl.allegro.tech.hermes.common.message.wrapper.MessageContentWrapper;
+import pl.allegro.tech.hermes.frontend.cache.topic.TopicsCache;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducer;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingException;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingResults;
+import pl.allegro.tech.hermes.frontend.publishing.message.JsonMessage;
+import pl.allegro.tech.hermes.frontend.publishing.message.Message;
+import pl.allegro.tech.hermes.frontend.publishing.message.MessageIdGenerator;
+
+import javax.inject.Inject;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static java.nio.charset.Charset.defaultCharset;
+
+public class MessagesPublishingStartupValidationHook implements ServiceAwareHook {
+
+    private static final Logger logger = LoggerFactory.getLogger(MessagesPublishingStartupValidationHook.class);
+    private final TopicsCache topicsCache;
+    private final MessageContentWrapper messageContentWrapper;
+    private final Clock clock;
+    private final ConfigFactory config;
+    private final BrokerMessagesProducer producer;
+    private final ScheduledExecutorService scheduler;
+
+    @Inject
+    public MessagesPublishingStartupValidationHook(ConfigFactory config,
+                                                   BrokerMessagesProducer brokerMessagesProducer,
+                                                   MessageContentWrapper messageContentWrapper,
+                                                   Clock clock,
+                                                   TopicsCache topicsCache) {
+        this.scheduler = Executors.newScheduledThreadPool(
+                2, new ThreadFactoryBuilder().setNameFormat("messages-publishing-startup-validation-%d").build()
```

Comment:
Maybe a little comment why 2 threads ? :) just like `required by retry policy to properly handle main operation and monitor its timeout conditions at the same time` ?